File: 2019/day05.cpp
#include <cstdio>
#include <vector>
#include "lib/intcode.cpp"

using namespace std;

int main() {
  vector<int> baseProg;
  readProg(baseProg);

  printf("%d\n", runProg(baseProg, {1}));
  printf("%d\n", runProg(baseProg, {5}));
  return 0;
}

File: 2019/day05.hs
import Control.Arrow
import Lib.Intcode

diagnostic :: Int -> Prog -> Int
diagnostic inp = runProgMonad $ do
  pushInput inp
  runProg
  lastOutput

solve1 :: Prog -> Int
solve1 = diagnostic 1

solve2 :: Prog -> Int
solve2 = diagnostic 5

main :: IO ()
main = interact $ show . (solve1 &&& solve2) . parseProg
